model/atmosphere/dycore/tests/dycore/integration_tests/test_solve_nonhydro.py

```diff
@@ -1745,6 +1746,305 @@ def test_apply_divergence_damping_and_update_vn(
+    z_graddiv_vn = savepoint_dycore_30_to_38_init.z_graddiv_vn()
+    vt = savepoint_dycore_30_to_38_init.vt()
+    mass_fl_e = savepoint_dycore_30_to_38_init.mass_fl_e()
+    z_theta_v_fl_e = savepoint_dycore_30_to_38_init.z_theta_v_fl_e()
+    z_vt_ie = savepoint_dycore_30_to_38_init.z_vt_ie()
+    vn_ie = savepoint_dycore_30_to_38_init.vn_ie()
+    z_kin_hor_e = savepoint_dycore_30_to_38_init.z_kin_hor_e()
+    z_w_concorr_me = savepoint_dycore_30_to_38_init.z_w_concorr_me()
+
+    e_flx_avg = interpolation_savepoint.e_flx_avg()
+    vn = savepoint_dycore_30_to_38_init.vn()
+    z_rho_e = savepoint_dycore_30_to_38_init.z_rho_e()
+    z_theta_v_e = savepoint_dycore_30_to_38_init.z_theta_v_e()
+
+    horizontal_start = icon_grid.start_index(edge_domain(h_grid.Zone.LATERAL_BOUNDARY_LEVEL_5))
+    horizontal_end = icon_grid.end_index(edge_domain(h_grid.Zone.HALO_LEVEL_2))
+
+    z_vn_avg_ref = savepoint_dycore_30_to_38_exit.z_vn_avg()
+    z_graddiv_vn_ref = savepoint_dycore_30_to_38_exit.z_graddiv_vn()
+    vt_ref = savepoint_dycore_30_to_38_exit.vt()
+    mass_fl_e_ref = savepoint_dycore_30_to_38_exit.mass_fl_e()
+    z_theta_v_fl_e_ref = savepoint_dycore_30_to_38_exit.z_theta_v_fl_e()
+    vn_ie_ref = savepoint_dycore_30_to_38_exit.vn_ie()
+    z_vt_ie_ref = savepoint_dycore_30_to_38_exit.z_vt_ie()
+    z_kin_hor_e_ref = savepoint_dycore_30_to_38_exit.z_kin_hor_e()
+    z_w_concorr_me_ref = savepoint_dycore_30_to_38_exit.z_w_concorr_me()
+
+    compute_horizontal_velocity_quantities.compute_horizontal_velocity_quantities_and_fluxes.with_backend(
+        backend
+    )(
+        spatially_averaged_vn=z_vn_avg,
+        horizontal_gradient_of_normal_wind_divergence=z_graddiv_vn,
+        tangential_wind=vt,
+        mass_flux_at_edges_on_model_levels=mass_fl_e,
+        theta_v_flux_at_edges_on_model_levels=z_theta_v_fl_e,
+        tangential_wind_on_half_levels=z_vt_ie,
+        vn_on_half_levels=vn_ie,
+        horizontal_kinetic_energy_at_edges_on_model_levels=z_kin_hor_e,
+        contravariant_correction_at_edges_on_model_levels=z_w_concorr_me,
+        vn=vn,
+        e_flx_avg=e_flx_avg,
+        geofac_grdiv=geofac_grdiv,
+        rbf_vec_coeff_e=rbf_vec_coeff_e,
+        rho_at_edges_on_model_levels=z_rho_e,
+        theta_v_at_edges_on_model_levels=z_theta_v_e,
+        ddqz_z_full_e=ddqz_z_full_e,
+        ddxn_z_full=ddxn_z_full,
+        ddxt_z_full=ddxt_z_full,
+        wgtfac_e=wgtfac_e,
+        wgtfacq_e=wgtfacq_e,
+        nflatlev=nflatlev,
+        horizontal_start=horizontal_start,
+        horizontal_end=horizontal_end,
+        vertical_start=0,
+        vertical_end=icon_grid.num_levels + 1,
+        offset_provider={
+            "E2C2EO": icon_grid.get_connectivity("E2C2EO"),
+            "E2C2E": icon_grid.get_connectivity("E2C2E"),
+            "Koff": dims.KDim,
+        },
+    )
+
+    assert helpers.dallclose(
+        z_vn_avg_ref.asnumpy()[:, :],
+        z_vn_avg.asnumpy()[:, :],
```

Comment:
I don't think you need to use indexing if you're not actually indexing anything
```suggestion
        z_vn_avg_ref.asnumpy().
        z_vn_avg.asnumpy(),
```